tests/unittests/ontology_class_tests.py

```
@@ -174,8 +175,274 @@ def test_to_dict_matches_when(self):
+    def test_dispatch_or(self):
+        d = {
+            "operator": "or",
+            "conditions": [
+                {"operator": "equals", "field": "a", "value": 1},
+            ],
+        }
+        cond = WhenCondition.from_dict(d)
+        self.assertIsInstance(cond, WhenOr)
+
+    def test_dispatch_unknown_operator_raises(self):
+        with self.assertRaises(ValueError):
```

Ruff (0.15.12)

[warning] Use `pytest.raises` instead of unittest-style `assertRaises`

Replace `assertRaises` with `pytest.raises`

(PT027)